app/signals/apps/llm_prediction/services/prediction.py

```diff
@@ -0,0 +1,154 @@
+def get_system_prompt() -> str:
+    categories = get_categories_with_description()
+
+    system_prompt = f"""
+        You are a classification assistant. Your task is to assign the most appropriate subcategory from the hierarchy. 
+        You may only choose a specific subcategory if the message clearly matches its description. 
+        If the message fits a parent category but not a specific subcategory, choose that parent’s “Overig” subcategory.
+        If the message does not match any parent category, choose:
+        "main_category": "Overig"
+        "sub_category": "Overig"
+        
+        ### Category Hierarchy
+        {format_categories(categories)}
+        
+        ### Classification Rules
+        1. Understand the message. Read the message carefully and infer its real-world meaning.
+        2. Choose exactly one subcategory. Never return multiple or top-level categories.
+        3. Base your choice strictly on meaning and the category descriptions. Do not rely only on keywords.
+        4. Only choose a specific subcategory if the message clearly and directly matches that subcategory’s definition. 
+           If the meaning does not strongly fit a specific subcategory, do NOT pick it.
+        5. If multiple subcategories seem possible, choose the one with the most specific and direct semantic match.
+        6. Never create, alter, or merge categories. Use only the categories exactly as listed.
+        7. If a message clearly belongs to a parent category but does not strongly match any of that parent’s specific subcategories, then assign the subcategory “Overig” for that parent. The resulting subcategory label must use the pattern:
+           "Overig <main_category>"
+        8. If the message does not clearly match any parent category at all, return:
+           "main_category": "Overig"
+           "sub_category": "Overig"
+        
+        ### Output Format
+        Return a single JSON object exactly like this:
+        {{
+          "main_category": "<main_category>",
+          "sub_category": "<sub_category>",
+          "text": "<original_message>"
+        }}
+        
+        ### Example
+        **Message:**
+        A car has crashed into a tree.
+        
+        **Expected Output:**
+        {{
+          "main_category": "Openbaar groen en water"
+          "sub_category": "Boom",
+          "text": "A car has crashed into a tree."
+        }}
+        
+        ### Message
+        {{message}}
+        """
+
+    return system_prompt
+
+
+def get_prediction(text: str) -> LlmResponse:
+    prompt = get_system_prompt()
+
+    client = get_openai_client()
+    response = client.chat.completions.create(
+        messages=[
+            {
+                "role": "system",
+                "content": prompt,
+            },
+            {
+                "role": "user",
+                "content": text,
+            },
+        ],
+        model='mistral-small-3.2-24b-instruct-2506',
```

Comment:
Het is mooi om hier een instelling van te maken via settings en een environment variabele, dan kunnen we dat makkelijk instellen. Dit mag wel de default zijn.